common/lib/xmodule/xmodule/error_module.py

```diff
@@ -30,62 +37,56 @@
 
 class ErrorFields(object):
     """
-    XBlock fields used by the ErrorModules
+    XBlock fields used by the ErrorBlocks
     """
     contents = String(scope=Scope.content)
     error_msg = String(scope=Scope.content)
     display_name = String(scope=Scope.settings)
 
 
-class ErrorModule(ErrorFields, XModule):
+class ErrorBlock(
+    ErrorFields,
+    XModuleDescriptorToXBlockMixin,
+    XModuleToXBlockMixin,
+    HTMLSnippet,
+    ResourceTemplates,
+    XModuleMixin,
+):
     """
     Module that gets shown to staff when there has been an error while
     loading or rendering other modules
     """
 
+    resources_dir = None
+
+    def student_view(self, _context):
+        """
+        Return a fragment that contains the html for the student view.
+        """
+        fragment = Fragment(self.get_html())
+        return fragment
+
     def get_html(self):
-        '''Show an error to staff.
-        TODO (vshnayder): proper style, divs, etc.
-        '''
+        """
+        Show an error to staff.
+        """
         # staff get to see all the details
         return self.system.render_template('module-error.html', {
             'staff_access': True,
             'data': self.contents,
             'error': self.error_msg,
         })
 
-
-class NonStaffErrorModule(ErrorFields, XModule):
-    """
-    Module that gets shown to students when there has been an error while
-    loading or rendering other modules
-    """
-    def get_html(self):
-        '''Show an error to a student.
-        TODO (vshnayder): proper style, divs, etc.
-        '''
-        # staff get to see all the details
-        return self.system.render_template('module-error.html', {
-            'staff_access': False,
-            'data': "",
-            'error': "",
-        })
-
-
-class ErrorDescriptor(ErrorFields, XModuleDescriptor):
-    """
-    Module that provides a raw editing view of broken xml.
-    """
-    module_class = ErrorModule
-    resources_dir = None
-
-    def get_html(self):
-        return u''
+    def studio_view(self, _context):
```

Comment:
Everything below this line is included in the `ErrorDescriptor` so even though the `NonStaffErrorModule` does not inherit from `ErrorModule`, all the methods below need to be included in the `NonStaffErrorBlock`.